### PR TITLE
example(rust): Create Rust example project

### PR DESCRIPTION
**Pull request by Arm1stice**
This PR adds a new Rust example project. It is built using Bazel with [rules_rust](https://github.com/bazelbuild/rules_rust) and dependencies are managed using [cargo-raze](https://github.com/google/cargo-raze) using remote dependency mode.

### CODE REVIEW

File: .bazelignore [hunk 1]
```diff
@@ -1 +1,2 @@
 node_modules
+kythe/rust/examples
```
Comment (zrlk) [inline .bazelignore:2]: nit (here and in other files): please add final newline

File: kythe/rust/examples/hello_world_vendored/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+```shell
+$ cargo install cargo-raze
+```
+Enter the `cargo` directory:
+```shell
+$ cd cargo
+```
+Vendor the dependencies:
+```shell
+$ cargo vendor --versioned-dirs --locked
```
Comment (zrlk) [inline kythe/rust/examples/hello_world_vendored/README.md:18]: This command fails for me:
```
error: Found argument '--versioned-dirs' which wasn't expected, or isn't valid in this context
```
I have cargo 1.39.0 installed; is this too old?

**Comment by Arm1stice** [comment]
Based on our conversation, I consolidated this PR into one Rust example. Currently, the WORKSPACE file links to my forked branch of the Bazel Rust rules while I am waiting for my PR to get merged.

**Comment by Arm1stice** [comment]
My bugfix in bazelbuild/rules_rust so I have updated the WORKSPACE file. We are no longer blocked on this PR.

File: WORKSPACE [hunk 3]
```diff
@@ -107,3 +107,31 @@ rbe_autoconfig(
...
+    name = "io_bazel_rules_rust",
+    sha256 = "fbdb7d0c88969fa6e71609bbbb20660e1cb7dcaa6ce84d8713c06bea5880cbb2",
+    strip_prefix = "rules_rust-6d00e2576688500dc857dd668cf1db1255b2976e",
+    urls = [
+        "https://github.com/bazelbuild/rules_rust/archive/6d00e2576688500dc857dd668cf1db1255b2976e.tar.gz",
+    ],
+)
+
+http_archive(
+    name = "bazel_skylib",
```
Comment (schroederc) [inline WORKSPACE:122]: Already exists in `setup.bzl`

File: WORKSPACE [hunk 4]
```diff
@@ -107,3 +107,31 @@ rbe_autoconfig(
     toolchain_config_suite_spec = DEFAULT_TOOLCHAIN_CONFIG_SUITE_SPEC,
     use_legacy_platform_definition = False,
 )
+
+# Pulls the latest Rust build rules as of 05/28/2020
+http_archive(
+    name = "io_bazel_rules_rust",
```
Comment (schroederc) [inline WORKSPACE:113]: Put rules in the `setup.bzl` file, following the pattern of the others.

**Comment by zrlk** [comment]
Looks like the commit message needs updated.

File: WORKSPACE [hunk 5]
```diff
@@ -107,3 +107,31 @@ rbe_autoconfig(
...
+)
+
+load("@io_bazel_rules_rust//rust:repositories.bzl", "rust_repositories")
+rust_repositories(version = "nightly", iso_date = "2020-05-15")
+
+load("@io_bazel_rules_rust//:workspace.bzl", "bazel_version")
+bazel_version(name = "bazel_version")
+
+load("//:external.bzl", "rust_dependencies")
+rust_dependencies()
```
Comment (schroederc) [inline WORKSPACE:137]: Put the rule config calls into the `_rule_dependencies()` macro in `external.bzl`

**Comment by Arm1stice** [comment]
Updated PR description per @zrlk's comment. Also made changes to address @schroederc's comments.

**Pull request merged**